### PR TITLE
Feature local variable

### PR DESCRIPTION
**Pull request by gabkk**

### CODE REVIEW

File: .deps/builtins/bg/builtin_bg.d [hunk 1]
```diff
@@ -0,0 +1,67 @@
+.objs/builtins/bg/builtin_bg.o: srcs/builtins/bg/builtin_bg.c \
```
Comment (jgigault) [inline .deps/builtins/bg/builtin_bg.d]: What's the fuck!!? 
+12,524 lines of code in this PR :/

Comment (gabkk) [reply .deps/builtins/bg/builtin_bg.d]: .deps ...

Comment (jgigault) [reply .deps/builtins/bg/builtin_bg.d]: Pro tip: Never use "git add -A" ;-)

File: srcs/job/job_launch.c [hunk 2]
```diff
@@ -12,7 +12,7 @@ static int			s_fork_it(t_sh *sh, t_job *j, t_proc *p)
 
 	if ((p->envp = ft_array_dup(sh->envp)) == NULL)
 		return (ST_MALLOC);
-	if ((ret = env_update_from_cmd_line(&p->argv, &p->argc, &p->envp)) != ST_OK)
+	if ((ret = env_or_var_update_from_cmd_line(&p, &sh)) != ST_OK)
```
Comment (jgigault) [inline srcs/job/job_launch.c:15]: here again, no need to have a pointer to a pointer to t_proc and t_sh structures :-)

Comment (jgigault) [reply srcs/job/job_launch.c:15]: here we need to be careful with the PR #57 

image

File: srcs/quoting/quoting_new_context.c [hunk 3]
```diff
@@ -61,7 +68,7 @@ static int				s_qloop(char *cmd,
 	buff_quote = NULL;
 	while ((parser(c->sh, cmd, F_PARSING_TERMCAPS, NULL)) != ST_OK)
 	{
-		(cmd[ft_strlen(cmd)-1] == '\n') ? (tmp = ft_strdup(cmd)) :
+		(cmd[ft_strlen(cmd) - 1] == '\n') ? (tmp = ft_strdup(cmd)) :
 			(tmp = ft_strjoin(cmd, "\n"));
 		termcaps_string_to_command_line(ft_strlen(cmd), cmd, &c->command_line);
```
Comment (jgigault) [inline srcs/quoting/quoting_new_context.c:73]: if (tmp == NULL)
return ST_MALLOC

**Comment by jgigault** [comment]
@gabkk leaks :  

```
echo "$HQSKGDHJSQGDK"  #variable not found
```

**Pull request closed without merging**